blocks/filter-list/filter-list.js

```diff
@@ -0,0 +1,185 @@
+import { readBlockConfig } from '../../scripts/lib-franklin.js';
+import ffetch from '../../scripts/ffetch.js';
+
+function buildCell(rowIndex) {
+  const cell = rowIndex ? document.createElement('td') : document.createElement('th');
+  if (!rowIndex) cell.setAttribute('scope', 'col');
+  return cell;
+}
+
+function dataMapLookup(dataMap, value) {
+  const foundValue = dataMap.find((item) => item.key === value);
+  return foundValue ? foundValue.value : value;
+}
+
+function populateTable(data, tableFields, dataMap, detailsUrl, dataSource, year, tbody) {
+  data.forEach((item, i) => {
+    const row = document.createElement('tr');
+    const isEvenRow = i % 2 === 0;
+    row.classList.add(isEvenRow ? 'row2' : 'row1');
+    tableFields.forEach(async (field) => {
+      const isDetailsLink = field.includes('(link)');
+      const rawValue = isDetailsLink ? item[field.replace('(link)', '').trim()] : item[field.trim()];
+      const value = dataMapLookup(dataMap, rawValue);
+      const cell = buildCell(1);
+      if (isDetailsLink) {
+        const a = document.createElement('a');
+        a.href = `${detailsUrl}?key=${item.Pkey}&dataSource=${dataSource}&year=${year}`;
+        a.innerText = value;
+        cell.append(a);
+      } else {
+        cell.innerText = value;
+      }
+      row.append(cell);
+    });
+    tbody.append(row);
+  });
+}
+
+async function loadMore(tableFields, dataMap, detailsUrl, dataSource, year, tbody, loadMoreDiv) {
+  const spreadsheet = `/data-source/${dataSource}/${dataSource}-data.json`;
+  const newData = await ffetch(spreadsheet).sheet(year).slice(20).all();
```

Comment:
Admittedly, I don't entirely understand what `all()` does, but I was recently given guidance to avoid it if there will potentially be a large number of records involved. The alternate way of doing this would be to use a `for ... of` loop:

```
const entries = ffetch(spreadsheet).sheet(year).slice(20);

// the linter won't like this line, but you can ignore the error
// eslint-disable-next-line no-restricted-syntax
for await (const entry of entries) {
  // maybe things could be slightly optimized by adding rows to the table
  // in here instead of storing everything in an array, then iterating over
  // the array again?
}
```

Reply:
i tried this loop before :S it wasn't working for me but i'll give it another try